topic-operator/src/main/java/io/strimzi/operator/topic/KubernetesHandler.java

```diff
@@ -0,0 +1,172 @@
+            TopicOperatorUtil.stopExternalRequestTimer(timerSample, metricsHolder::addFinalizerTimer, config.enableAdditionalMetrics(), config.namespace());
+            LOGGER.traceCr(reconcilableTopic.reconciliation(), "Added finalizer {}, resourceVersion now {}", FINALIZER_STRIMZI_IO_TO, TopicOperatorUtil.resourceVersion(withFinalizer));
+            return withFinalizer;
+        }
+        return reconcilableTopic.kt();
+    }
+
+    /**
+     * Remove finalizer from a KafkaTopic resource.
+     *
+     * @param reconcilableTopic Reconcilable topic.
+     * @return KafkaTopic resource.
+     */
+    public KafkaTopic removeFinalizer(ReconcilableTopic reconcilableTopic) {
+        if (reconcilableTopic.kt().getMetadata().getFinalizers().contains(FINALIZER_STRIMZI_IO_TO)) {
+            LOGGER.traceCr(reconcilableTopic.reconciliation(), "Removing finalizer {}", FINALIZER_STRIMZI_IO_TO);
+            var timerSample = TopicOperatorUtil.startExternalRequestTimer(metricsHolder, config.enableAdditionalMetrics());
+            var withoutFinalizer = Crds.topicOperation(kubernetesClient).resource(reconcilableTopic.kt()).edit(old ->
+                new KafkaTopicBuilder(old).editOrNewMetadata().removeFromFinalizers(FINALIZER_STRIMZI_IO_TO).endMetadata().build());
+            TopicOperatorUtil.stopExternalRequestTimer(timerSample, metricsHolder::removeFinalizerTimer, config.enableAdditionalMetrics(), config.namespace());
+            LOGGER.traceCr(reconcilableTopic.reconciliation(), "Removed finalizer {}, resourceVersion now {}", FINALIZER_STRIMZI_IO_TO, TopicOperatorUtil.resourceVersion(withoutFinalizer));
+            return withoutFinalizer;
+        } else {
+            return reconcilableTopic.kt();
+        }
+    }
+
+    /**
+     * Update the KafkaTopic status.
+     * 
+     * @param reconcilableTopic Reconcilable topic.
+     * @return KafkaTopic resource.
+     */
+    public KafkaTopic updateStatus(ReconcilableTopic reconcilableTopic) {
+        var oldStatus = Crds.topicOperation(kubernetesClient)
+            .inNamespace(reconcilableTopic.kt().getMetadata().getNamespace())
+            .withName(reconcilableTopic.kt().getMetadata().getName()).get().getStatus();
+        if (statusChanged(reconcilableTopic.kt(), oldStatus)) {
+            // the observedGeneration is initialized to 0 when creating a paused topic (oldStatus null, paused true)
+            // this will result in metadata.generation: 1 > status.observedGeneration: 0 (not reconciled)
+            reconcilableTopic.kt().getStatus().setObservedGeneration(reconcilableTopic.kt().getStatus() != null && oldStatus != null
+                ? !TopicOperatorUtil.isPaused(reconcilableTopic.kt()) ? reconcilableTopic.kt().getMetadata().getGeneration() : oldStatus.getObservedGeneration()
+                : !TopicOperatorUtil.isPaused(reconcilableTopic.kt()) ? reconcilableTopic.kt().getMetadata().getGeneration() : 0L);
```

Comment:
`!TopicOperatorUtil.isPaused(reconcilableTopic.kt())` appears in both branches. Can we simplify it?

Reply:
Yeah, this statement is confusing.We always need to update the observedGeneration, so we just have to address the special case of a new unamanged/paused topic by setting observedGeneration to zero. This is done in @10481.